File: poet-spring-boot-starter/src/main/java/com/skypyb/poet/spring/boot/core/util/StreamUtil.java
package com.skypyb.poet.spring.boot.core.util;

import java.io.Closeable;
import java.io.IOException;
import java.util.Arrays;
import java.util.Objects;

public class StreamUtil {

    public static void close(Closeable closeable) {
        if (Objects.isNull(closeable)) return;

        try {
            closeable.close();
        } catch (IOException e) {
            //忽略
        }
    }

    public static void close(Closeable... closeable) {
        if (Objects.isNull(closeable)) return;
        Arrays.stream(closeable).forEach(StreamUtil::close);
    }
}

File: poet-core/src/main/kotlin/com/skypyb/poet/core/util/StreamUtil.kt
package com.skypyb.poet.core.util

import java.io.Closeable
import java.io.IOException


class StreamUtil {

    companion object {
        @JvmStatic
        fun close(closeable: Closeable) {
            if (closeable == null) return
            try {
                closeable.close()
            } catch (e: IOException) { //忽略
            }
        }

        @JvmStatic
        fun close(vararg closeable: Closeable) {
            if (closeable == null) return
            closeable.forEach { close(it) }
        }
    }

}